### PR TITLE
Split bypass prerequisites

### PR DESCRIPTION
**Pull request by Separius**
## Summary

  This is PR 1 of 3 in the Puzzletron bypass/local-distillation stack.

  This PR contains prerequisite infrastructure only. It does not wire bypass distillation into the Puzzletron pipeline yet.

  Stack:
  1. **This PR**: shared prerequisites
  2. `ssameni/puzzletron-bypass-2-core`: bypass distillation core
  3. `ssameni/puzzletron-bypass-3-integration`: Puzzletron integration, configs, docs, GPU coverage

  ## What Changed

  - Added `ModelDescriptor.pruning_mixins()` so model families can expose pruning mixins needed by downstream bypass initialization.
  - Added KV-head pruning mixin support for GPT-OSS, Nemotron-H, Nemotron-H-v2, and Qwen3-VL descriptors.
  - Improved pruning utilities for nested language-model configs and missing attention bias config fields.
  - Added `create_train_dataloader()` and streaming-safe shuffle handling.
  - Added chat-template fallback for base models without `tokenizer.chat_template`.
  - Added Sewing Kit loss/helper exports needed by the later bypass core.
  - Updated child-state initialization to support composing multiple pruning mixins.
  - Updated warmup-step resolver to account for gradient accumulation.

  ## Why

  The bypass distillation MR needs these reusable pieces, but they are independently reviewable and useful without adding the bypass
  training stage itself.

  Splitting them out keeps the bypass core PR focused on the actual local-distillation engine.

  ## Tests

  Added focused unit coverage for:

  - Dataloader behavior
  - Bypass loss helpers
  - KV-head pruning utilities
  - Sewing Kit activity/input/function/needle behavior

## Summary by CodeRabbit

* **New Features**
  * KV-head pruning added for multiple model families; generic pruning mixin hook available.
  * New training dataloader factory for infinite, block-sized training streams.
  * Vectorwise and batched normalized MSE loss utilities.

* **Improvements**
  * Loss reports now show Δ-from-initial and visual indicators.
  * Chat-sample preprocessing tolerates tokenizers without chat templates.
  * More robust head-dimension/bias handling and grad-accum-aware warmup resolver.

* **Tests**
  * Extensive unit tests added across dataloaders, losses, pruning, hydra utils, and sewing-kit.

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/Model-Optimizer/pull/1468)

### CODE REVIEW

**Comment by Separius** [comment]
@AAnoosheh and @kevalmorabia97 ready for review (split the bypass MR into 3, this is the first one, nothing too important, just some preparations and tiny fixes)

**Comment by Separius** [comment]
@AAnoosheh ready for review

File: modelopt/torch/puzzletron/pruning/pruning_utils.py [hunk 1]
```diff
@@ -66,6 +67,30 @@ class HiddenSizeInitMode(Enum):
     CopyAsIs = "CopyAsIs"
 
 
+def _lm_attrs(config):
```
Comment (chochowski) [inline modelopt/torch/puzzletron/pruning/pruning_utils.py]: Can we use the descriptor function instead  `def get_language_model_config(config)`

Comment (Separius) [reply modelopt/torch/puzzletron/pruning/pruning_utils.py]: fair, fixed

**Comment by Separius** [comment]
@kevalmorabia97 ready for review

**Review by kevalmorabia97 — APPROVED** [review]

**Comment by kevalmorabia97** [comment]
/ok to test [e567f57](https://github.com/NVIDIA/Model-Optimizer/pull/1468/commits/e567f57328b18ec7127f7bd0f022e93eff54eccc)

**Review by chochowski — APPROVED** [review]
